### PR TITLE
Fix Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate output

### LINKED ISSUE #7905: Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate does not reset to zero each timestep
Issue overview
--------------
From user:

> According to the description in the input/output reference, I would expect that the following two outputs report same amount of generated CO2:
> 
> * Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s]
>  * Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s]
> However Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate reports higher CO2 gains compared to the first output mentioned. Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s] output corresponds to the input value described above (0.0001125 m3/s). While Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s] increases cumulatively over time

Sure enough, `ZnRpt(NZ).CO2Rate` is not getting zeroed [here](https://github.com/NREL/EnergyPlus/blob/be5b67255957ed53c59d16306878482e3b8634ad/src/EnergyPlus/InternalHeatGains.cc#L5261-L5268) every time step before is accumulates across a CO2 sources in the zone. Will push a fix shortly.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus9.3.0
 - Helpdesk ticket number 15425

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #7905 
 - Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate output was not getting reset to zero for each timestep, so the reported value kept accumulating over the course of the simulation.

The #7905 defect file should report exactly the same values in columns B and C:
CO2PEOPLE:Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s](TimeStep)
CLASSROOM:Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate [m3/s](TimeStep)

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
        *No diffs expected, because Zone Contaminant Source or Sink CO2 Gain Volume Flow Rate is not used in any of the testfiles.*
        *Added outputs to HeatPumpIAQP_GenericContamControl.idf*
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] ~If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label~
 - [ ] ~If structural output changes, add to output rules file and add OutputChange label~
 - [ ] ~If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies~

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] ~If branch is behind develop, merge develop and build locally to check for side effects of the merge~
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] ~Check any new function arguments for performance impacts~
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

### CODE REVIEW

File: src/EnergyPlus/InternalHeatGains.cc [hunk 1]
```diff
@@ -5265,6 +5265,7 @@ namespace InternalHeatGains {
             e.HWPower = 0.0;
             e.SteamPower = 0.0;
             e.BaseHeatPower = 0.0;
+            e.CO2Rate = 0.0;
```
Comment (mjwitte) [inline src/EnergyPlus/InternalHeatGains.cc:5268]: This was forgotten here. Note that not all variables in `ZnRpt` are set to zero here. Just the ones that are accumulated with `+=` in this function.

**Comment by mjwitte** [comment]
Note the increase in unit test coverage.
```
Develop:                         Line Coverage         Function Coverage
InternalHeatGains.cc |   |   | 55.7 % | 2089 / 3751 | 76.5 % | 26 / 34
This branch:
InternalHeatGains.cc |   |   | 72.8 % | 2733 / 3752 | 76.5 % | 26 / 34
```

**Comment by rraustad** [comment]
No example files use this report variable? Should add it to one or more?

**Comment by rraustad** [comment]
Looked over the function where this change was made. This change looks like it includes everything.

**Comment by rraustad** [comment]
CO2 report looks correct now.

image

**Comment by rraustad** [comment]
Unit tests run to completion locally.

**Comment by rraustad** [comment]
The 1 unit test failure appears unrelated and this specific failure is occurring more often.

**Comment by rraustad** [comment]
@mjwitte this all looks good and is ready to merge. What do you think about adding the report to one of the example files?

File: tst/EnergyPlus/unit/InternalHeatGains.unit.cc [hunk 2]
```diff
@@ -829,3 +830,176 @@ TEST_F(EnergyPlusFixture, InternalHeatGains_CheckThermalComfortSchedules)
...
+    InternalHeatGains::InitInternalHeatGains();
+
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).LtsPower, 100.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).ElecPower, 150.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).GasPower, 200.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).HWPower, 250.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).SteamPower, 300.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).BaseHeatPower, 1500.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).CO2Rate, 0.0001125);
+}
```
Comment (rraustad) [inline tst/EnergyPlus/unit/InternalHeatGains.unit.cc:1005]: Why isn't this showing that little error with no line feed after the last bracket?

Comment (rraustad) [reply tst/EnergyPlus/unit/InternalHeatGains.unit.cc:1005]: Oh, because it's not showing line 1006.

**Comment by mjwitte** [comment]
@rraustad Added output variables to one test file and corrected the docs for some of the output variables.

**Review by rraustad — APPROVED** [review]
- Reports added to example file
 - Review doc changes, built IO Ref and reviewed changes
 - CI reports show expected results

**Pull request merged**